### PR TITLE
Load grayscale images using opencv

### PR DESCRIPTION
**Pull request by kukreja-vikramaditya**
See comments in code

### CODE REVIEW

File: BITS-DIC.pro [hunk 1]
```diff
@@ -26,3 +26,6 @@ DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0x060000    # disables all the APIs depr
 include(src/headers/headers.pri)
 include(src/sources/sources.pri)
 include(src/forms/forms.pri)
+
+INCLUDEPATH += /usr/local/include/opencv
```
Comment (Rushi98) [inline BITS-DIC.pro:30]: This will work only on unix machines.

File: src/sources/mainwindow.cpp [hunk 2]
```diff
@@ -3,35 +3,78 @@
...
+#include <QDesktopWidget>
+#include <opencv2/opencv.hpp>
 
 MainWindow::MainWindow(QWidget *parent) :
     QMainWindow(parent),
     ui(new Ui::MainWindow)
 {
     ui->setupUi(this);
     this->setWindowTitle(WINDOW_TITLE);
+    //this->showMaximized();
```
Comment (Rushi98) [inline src/sources/mainwindow.cpp:15]: we can keep this

File: src/sources/mainwindow.cpp [hunk 3]
```diff
@@ -3,35 +3,78 @@
...
+        //this code is buggy if image has very high aspect ratio or very low i think
+        //example would be sceenshot of a dual monitor setup
+        //we may not come across such examples tho. so it should be fine
+        int width = convertedImage.width() % 512;
+        int height = convertedImage.height() % 512;
+
+        int minSize = std::min(width, height);
+        minSize = 512 / minSize;
+        height *= minSize;
+        width *= minSize;
```
Comment (Rushi98) [inline src/sources/mainwindow.cpp:71]: This is incorrect.
What if image is 512 x 800 ?

**Review by Rushi98 — CHANGES_REQUESTED** [review]
I think the scaling down code is wrong.
Windows version for opencv can be postponed for now.

File: src/sources/mainwindow.cpp (hunk 3 above)
Comment (Rushi98) [reply src/sources/mainwindow.cpp:71]: We're scaling down so that max dimension is 512 right?

**Comment by iniyanakila** [comment]
yes please try to finish the tasks first ... hope you can give me an executable like before.
I would like to compare the 8-bit version of the reference image with Ncorr output.
I would also like to see the pixel-by-pixel gradients and B-spline coefficients before we proceed further to the next step.

Cheers
Iniyan

File: src/sources/mainwindow.cpp (hunk 3 above)
Comment (iniyanakila) [reply src/sources/mainwindow.cpp:71]: why exactly do we need to scale down?

Comment (Rushi98) [reply src/sources/mainwindow.cpp:71]: @iniyanakila Sir, This is so that we could see full portion of large image on screen.

**Comment by Rushi98** [comment]
@kukreja-vikramaditya see f015c7c05d93c8c41588996c98f74a2ecbb82900 for corrected code.

File: src/sources/mainwindow.cpp (hunk 3 above)
Comment (iniyanakila) [reply src/sources/mainwindow.cpp:71]: oh, it is for visualizing the image in the window... then its fine. If you can define a region of interest (ROI) by drawing a rectangle in the reference image and if you can convert the ROI into its 8-bit version, then you immediately proceed to finding intensity gradients and B-spline coefficients for the defined ROI.

PS. If there are any difficult parts that are not important to proceed further, we can come back and write code for that later. Prioritize whenever required.

cheers
Iniyan

**Pull request closed without merging**